cl/recap/management/commands/pacer_iquery_scraper.py

```diff
@@ -0,0 +1,81 @@
+# coding=utf-8
+from __future__ import print_function
+
+import json
+import re
+
+import requests
+from django.conf import settings
+
+from cl.alerts.models import DocketAlert
+from cl.celery import app
+from cl.lib.command_utils import VerboseCommand
+from cl.lib.pacer import map_cl_to_pacer_id
+from cl.lib.pacer_session import get_or_cache_pacer_cookies
+from cl.recap.mergers import (
+    update_docket_metadata,
+    add_bankruptcy_data_to_docket,
+)
+from cl.search.models import Docket
+from cl.search.tasks import add_items_to_solr
+from juriscraper.pacer import CaseQuery, PacerSession
+
+
+def get_dockets():
+    visits = requests.get(
+        settings.MATOMO_REPORT_URL,
+        timeout=1,
+        params={
+            "idsite": settings.MATOMO_SITE_ID,
+            "module": "API",
+            "method": "Live.getLastVisitsDetails",
+            "period": "week",
+            "format": "json",
+            "date": "today",
+            "token_auth": settings.MATOMO_TOKEN,
+        },
+    )
+    visitsjson = visits.json()
+    urllist = [
+        item["actionDetails"][0]["url"]
+        if "url" in item["actionDetails"][0]
+        else ""
+        for item in visitsjson
+    ]
+    urllistasstring = "|".join(urllist)
+    docket_list = set(re.findall("/docket/([0-9]+)/", urllistasstring))
```

Comment:
What about doing it this way:

```suggestion
    docket_ids = set()
    for item in visits.json():
        url = item["actionDetails][0].get("url")
        if url is None:
            continue
        docket_ids.update(re.findall("/docket/[0-9]+/", url))
```

I don't know if performance will be much different, but I suspect it's close enough not to matter too much. In the meantime, it's a lot more readable. (No list comprehension, no empty else statement, no pipejoining and regexing...)

Reply:
It also looks like `actionDetails` is a list that can have lots of entries:

```
 u"actionDetails": [
            {
                u"generationTimeMilliseconds": u"3611",
                u"pageId": u"24074601",
                u"title": u"Yang v. New York State Board of Electi, 20-1494 \u2013 CourtListener.com",
                u"url": u"https://www.courtlistener.com/docket/28228/yang-v-new-york-state-board-of-electi/",
                u"timestamp": 1589500858,
                u"idpageview": u"t7DYwJ",
                u"pageTitle": u"Yang v. New York State Board of Electi, 20-1494 \u2013 CourtListener.com",
                u"pageIdAction": u"6647911",
                u"serverTimePretty": u"May 15, 2020 00:00:58",
                u"timeSpent": u"3",
                u"iconSVG": u"plugins/Morpheus/images/action.svg",
                u"subtitle": u"https://www.courtlistener.com/docket/28228/yang-v-new-york-state-board-of-electi/",
                u"interactionPosition": u"1",
                u"generationTime": u"3.61s",
                u"type": u"action",
                u"timeSpentPretty": u"3s",
                u"icon": u"",
            },
            {
                u"generationTimeMilliseconds": u"5121",
                u"pageId": u"24074602",
                u"title": u"United States v. FLYNN, 1:17-cr-00232 \u2013 CourtListener.com",
                u"url": u"https://www.courtlistener.com/docket/6234142/united-states-v-flynn/?page=2",
                u"timestamp": 1589500861,
                u"idpageview": u"jEdctL",
                u"pageTitle": u"United States v. FLYNN, 1:17-cr-00232 \u2013 CourtListener.com",
                u"pageIdAction": u"3310804",
                u"serverTimePretty": u"May 15, 2020 00:01:01",
                u"timeSpent": u"68",
                u"iconSVG": u"plugins/Morpheus/images/action.svg",
                u"subtitle": u"https://www.courtlistener.com/docket/6234142/united-states-v-flynn/?page=2",
                u"interactionPosition": u"2",
                u"generationTime": u"5.12s",
                u"type": u"action",
                u"timeSpentPretty": u"1 min 8s",
                u"icon": u"",
            },
```

Probably we should iterate over them all?

Reply:
Done